src/components/signon/SignOn.tsx

```diff
@@ -512,6 +523,19 @@ const SignOn = ({ navigation }: SignOnProps) => {
     )
   }
 
+  const ForgotPasswordButton = () => {
```

Comment:
looks like this was already a pattern in this file, but we should avoid creating components in the render body of other components. this actually leads to a big perf hit as react can't cache these components. 

some options we can do are:
- define these outside the signon component
- make these inline functions. to do that, we should use camelCase, and probably change the name to include render, and doing `{renderForgotPasswordButton()}` instead of `<ForgotPasswordButton />`
- we just make this a constant with `const forgotPasswordButton = (<Text ... />)`

Reply:
Yeah makes sense, i think the 2nd option is what the general codebase patterns follow so I'll update to that